ledger/src/blocktree_processor.rs

```diff
@@ -590,6 +575,34 @@ fn process_pending_slots(
     Ok(fork_info)
 }
 
+// Processes and replays the contents of a single slot, returns Error
+// if failed to play the slot
+fn process_single_slot(
+    blocktree: &Blocktree,
+    bank: &Arc<Bank>,
+    last_entry_hash: &Hash,
+    opts: &ProcessOptions,
+) -> result::Result<(), BlocktreeProcessorError> {
+    let slot = bank.slot();
+
+    // Fetch all entries for this slot
+    let entries = blocktree.get_slot_entries(slot, 0, None).map_err(|err| {
+        warn!("Failed to load entries for slot {}: {:?}", slot, err);
+        BlocktreeProcessorError::FailedToLoadEntries
+    })?;
+
+    // If this errors with a fatal error, should mark the slot as dead so
+    // validators don't replay this slot and  see DuplicateSignature errors
+    // later in ReplayStage
+    verify_and_process_slot_entries(&bank, &entries, *last_entry_hash, opts).map_err(|err| {
+        warn!("slot {} failed to verify: {}", slot, err);
```

Comment:
@jstarry This is where having the replay and blocktree processor verification be the same would be useful. 

If we don't mark this slot as dead here on failure, then ReplayStage will try to replay this slot again and will run into DuplicateSignature errors.

If they were the same implementation, then we know that if this fails here, it will also fail in ReplayStage, so it would be safe to mark the slot as dead here. Otherwise, if the implementations/criteria differ, one validator might mark one as dead, another won't depending on if they replay here on boot or in ReplayStage.